dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/renderer/service/RendererSettingsImpl.java

```diff
@@ -0,0 +1,19 @@
+package org.dashbuilder.renderer.service;
+
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.jboss.errai.bus.server.annotations.Service;
+
+@Service
+@ApplicationScoped
+public class RendererSettingsImpl implements RendererSettings {
+
+    String DEFAULT_RENDERER_PROPERTY = "org.dashbuilder.renderer.default";
+    
+    @Override
+    public String userDefaultRenderer() {
+        return System.getProperty(DEFAULT_RENDERER_PROPERTY);
```

Comment:
Thanks for your review.

I was expecting it to indeed return null, I tried Optional, but Errai remote does not seem to work with Optional at the moment.  however, I can change to return some specific value, such as Google Chart UUID.

The issue with returning a default value or using an enum is that Renderers are decoupled from the main API, it is discovered using CDI so it give us the flexibility to plug/unplug renderers. So the renderer API should not know any specific renderer.

What you think about these two actions to avoid NPEs:

1) Return by default an empty String;
2) Comment the method saying that it returns an empty String if no value is set by the users.

Thanks!

Reply:
@jhrcek Please let me know if you think that returning an empty String is a good approach and I can modify it tomorrow.
I will work back on this PR once https://github.com/kiegroup/appformer/pull/573 is merged

Thanks!

Reply:
Sorry, I missed your comment 2 days ago. Maybe I didn't receive email notification, because you didn't tag me in the comment?

> I tried Optional, but Errai remote does not seem to work with Optional at the moment.

Is that because Optional is not serializable? 

Empty String + nullcheck / check for empty string is fine with me.